lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -169,10 +169,131 @@ emitIndices(Location loc, RewriterBase &rewriter, const TargetInfoBase &target,
   return ret;
 }
 
+namespace {
+
+Value getSmemVecAddr(RankedTensorType registerTy,
```

Comment:
Perhaps a non-issue and could be investigated in a different PR as LLVM may be doinga  good work already optimising the generated code, but at the moment this function is doing plenty of non-trivial work and it's being called in a loop. Chances are we can factor out quite a bit of this work outside of the loop, and just leave the loop-dependent work inside. For example, all the sub-view lifting is loop-independent.